modules/gapi/include/opencv2/gapi/gtransform.hpp

```diff
@@ -0,0 +1,98 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+#ifndef OPENCV_GAPI_GTRANSFORM_HPP
+#define OPENCV_GAPI_GTRANSFORM_HPP
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+#include <opencv2/gapi/gcommon.hpp>
+#include <opencv2/gapi/util/util.hpp>
+#include <opencv2/gapi/garg.hpp>
+#include <opencv2/gapi/gtype_traits.hpp>
+#include <opencv2/gapi/util/compiler_hints.hpp>
+
+namespace cv
+{
+
+struct GAPI_EXPORTS GTransform
+{
+    using F = std::function<GArgs(const GArgs &)>;
+
+    const char *description;
+    F pattern;
+    F substitute;
+
+    explicit GTransform(const char *d, const F &p, const F &s) : description(d), pattern(p), substitute(s){};
+};
+
+namespace detail
+{
+
+template <typename, typename, typename>
+struct TransHelper;
+
+template <typename K, typename... Ins, typename Out>
+struct TransHelper<K, std::tuple<Ins...>, Out>
+{
+    template <typename Callable, int... IIs, int... OIs>
+    static GArgs get_impl(Callable f, const GArgs &in_args, detail::Seq<IIs...>, detail::Seq<OIs...>)
+    {
+        const auto r = detail::tuple_wrap_helper<Out>::get(f(in_args.at(IIs).template get<Ins>()...));
+        return GArgs{GArg(std::get<OIs>(r))...};
+    }
+
+    static GArgs get_pattern(const GArgs &in_args)
+    {
+        return get_impl(K::pattern, in_args, typename detail::MkSeq<sizeof...(Ins)>::type(),
+                        typename detail::MkSeq<std::tuple_size<typename detail::tuple_wrap_helper<Out>::type>::value>::type());
+    }
+    static GArgs get_substitute(const GArgs &in_args)
+    {
+        return get_impl(K::substitute, in_args, typename detail::MkSeq<sizeof...(Ins)>::type(),
+                        typename detail::MkSeq<std::tuple_size<typename detail::tuple_wrap_helper<Out>::type>::value>::type());
```

Comment:
`TransHelper` is already in `detail` namespace, all `detail::` stuff above can be removed

Reply:
Removed